util/kustomize/kustomize.go

```diff
@@ -124,7 +124,11 @@ func (k *kustomize) Build(opts *v1alpha1.ApplicationSourceKustomize, kustomizeOp
 			if opts.ForceCommonLabels {
 				args = append(args, "--force")
 			}
-			cmd := exec.Command(k.getBinaryPath(), append(args, mapToEditAddArgs(opts.CommonLabels)...)...)
+			commonLabels := map[string]string{}
+			for name, value := range opts.CommonLabels {
+				commonLabels[name] = envVars.Envsubst(value)
```

Comment:
Suppose someone is already using a `commonAnnotation` with a $ in it, e.g. `$ilikedollarsigns`, and they expect it to be interpreted literally... will this cause their annotations to be replaced with empty strings?

Reply:
We could toggle the behavior on a new field, e.g. `commonLabelsEnvsubst`. I don't like adding a new field, but I really don't wanna mess up folks' labels.

Reply:
Oh wait, these are labels! They can't have dollar signs anyway: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Reply:
Ah, but we would still need a toggle for annotations.

Reply:
@AndrewChubatiuk apologies for the lack of response. I had to prioritize some high-priority stuff for 2.6. Let me target this for 2.7.